packages/react-components/react-switch/src/components/Switch/useSwitchStyles.ts

```diff
@@ -69,19 +68,30 @@ const useIndicatorStyles = makeStyles({
       },
     },
   },
+
+  before: {
+    ...shorthands.margin(tokens.spacingVerticalS, tokens.spacingHorizontalS, tokens.spacingVerticalS, 0),
+  },
+  after: {
+    ...shorthands.margin(tokens.spacingVerticalS, 0, tokens.spacingVerticalS, tokens.spacingHorizontalS),
+  },
+  above: {
+    ...shorthands.margin(0, tokens.spacingHorizontalS, tokens.spacingVerticalS, tokens.spacingHorizontalS),
+  },
 });
 
 const useInputStyles = makeStyles({
   base: {
     boxSizing: 'border-box',
     cursor: 'pointer',
     height: '100%',
-    left: 0,
     ...shorthands.margin(0),
     opacity: 0,
     position: 'absolute',
-    top: 0,
-    width: '100%',
+
+    // Calculate the width of the hidden input by taking into account the size of the indicator + the padding around it.
+    // This is done so that clicking on that "empty space" still toggles the switch.
+    width: `calc(${trackWidth}px + ${tokens.spacingHorizontalS})`,
```

Comment:
I haven't read the switch styles in as much detail, but similar to my comment here in checkbox, does this still need `spacingHorizontalS` included in it?
```suggestion
    width: `${trackWidth}px`,
```

Reply:
Yup, similar reason to the one above.